### ISSUE TITLE
Required options

### ISSUE THREAD

**Issue by scottcorgan** [desiringgod/components-rails#2]
### Idea

---

@joeosburn-desiringgod recommended I open these ideas here.

It's pretty helpful to be able to declare some options to be passed into the component as required. By default, they would be optional. This helps ensure that the errors don't send us in the wrong direction. This especially happens when we have nested components.

Example:

```erb
<%= icon(type: 'arrow') %>
```

Here, type is required because anything else doesn't make sense (there is no default value for this one). Display some sort of error like "The component 'icon' requires type to be defined" (or something similar) would make our components easier to use and learn.

**Comment by joeosburn-desiringgod** [comment]
I'm good with implementing this.

@benhutton Are you good with this? I'm about to dive into implementing this.

**Comment by benhutton** [comment]
@joeosburn-desiringgod yeah. Seems like it should be done in conjunction with #5, or at least with the API for #5 in mind, right?

**Comment by joeosburn-desiringgod** [comment]
@benhutton Yes! Though, at this point, I'm thinking the implementation of the two will be very different and probably unrelated. However, I'll get to that once I have a firm idea in mind.
